validated/dash_apps/finished_apps/daily_validation.py

```diff
@@ -84,15 +145,205 @@
+        detail_add_clicks (int): Number of times detail-add-button was clicked
+        in_daily_selected_rows (list[dict]): Selected rows in table_daily
+        in_detail_selected_rows (list[dict]): Selected rows in table_detail
+        in_detail_row_data (list[dict]): Full row data for table_detail
+
+    Returns:
+        tuple[str, str, go.Figure, list[dict], list[dict], dict, dict]:
+            Callback outputs
+    """
+    global DATA_DAILY, DATA_DETAIL
+
+    ctx = dash.callback_context
+    if not ctx.triggered:
+        return (dash.no_update,) * 7
+    button_id = ctx.triggered[0]["prop_id"].split(".")[0]
+
+    out_daily_status = dash.no_update
+    out_detail_status = dash.no_update
+    out_plot = dash.no_update
+    out_daily_row_data = dash.no_update
+    out_detail_row_data = dash.no_update
+    out_detail_row_transaction = dash.no_update
+    out_detail_scroll = dash.no_update
+
+    daily_refresh_required = False
+    detail_refresh_required = False
+    plot_refresh_required = False
+
+    # Button: Daily save
+    if button_id == "daily-save-button" and in_daily_selected_rows is not None:
+        conditions = get_conditions(in_daily_selected_rows)
+        save_to_validated(
+            variable=variable,
+            station=station,
+            to_delete=conditions,
+            start_date=start_date,
+            end_date=end_date,
+            minimum=minimum,
+            maximum=maximum,
+        )
+        out_daily_status = f"{len(in_daily_selected_rows)} days saved to Validated"
+        plot_refresh_required = True
+        daily_refresh_required = True
+
+    # Button: Daily reset
+    elif button_id == "daily-reset-button":
+        reset_daily_validated(
+            variable=variable,
+            station=station,
+            start_date=start_date,
+            end_date=end_date,
+        )
+        out_daily_status = "Validation reset"
+        plot_refresh_required = True
+        daily_refresh_required = True
+
+    # Button: Detail save
+    elif button_id == "detail-save-button" and in_detail_selected_rows is not None:
+        selected_ids = {row["id"] for row in in_detail_selected_rows}
+        for row in in_detail_row_data:
+            row["is_selected"] = row["id"] in selected_ids
+        save_detail_to_validated(
+            data_list=in_detail_row_data,
+            variable=variable,
+            station=station,
+        )
+        out_detail_status = f"{len(in_detail_selected_rows)} entries saved to Validated"
+        plot_refresh_required = True
+        detail_refresh_required = True
+
+    # Button: Detail reset
+    elif button_id == "detail-reset-button":
+        reset_detail_validated(
+            data_list=in_detail_row_data,
+            variable=variable,
+            station=station,
+        )
+        out_detail_status = "Validation reset"
+        plot_refresh_required = True
+        detail_refresh_required = True
```

Comment:
When clicking this, part of the validated plot disapears, but nothing to do with the selected/unselected range.

Reply:
This is by design, as the reset buttons are intended to remove the entire table from validated, rather than just the selected rows (so if you mess up and want to start again). I could add a separate "Remove from Validated" button to remove just the selected rows if helpful?